### PR TITLE
Enabled 'Shopping Cart' tab for customer edit interface in admin

### PR DESCRIPTION
**Pull request by rav-redchamps**
### Fixed Issues (if relevant)

1. magento/magento2#6287: Customer Admin Shopping Cart View Missing

### Manual testing scenarios (*)

1. Navigated to 'Customers > All Customers > Edit Specific Customer' in admin and verified the visibility of 'Shopping Cart' tab and its content grid 
2. Added few products to customer's cart in frontend and verified their visibility in 'Shopping Cart' tab
3. Tested product 'Delete' and 'Configure' actions in the tab

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=20918) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @rav-redchamps. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by rav-redchamps** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rav-redchamps. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rav-redchamps, here is your [new Magento instance](https://pr-20918.instances.magento-community.engineering).
Admin access: [https://pr-20918.instances.magento-community.engineering/admin](https://pr-20918.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by rav-redchamps** [comment]
@magento-engcom-team seems like the Travis automated tests stucked. It is running for more than 24 hours. Kindly check.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-4126](https://jira.corp.magento.com/browse/ENGCOM-4126) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@rav-redchamps thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by sivaschenko** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, here is your [new Magento instance](https://pr-20918.instances.magento-community.engineering).
Admin access: [https://pr-20918.instances.magento-community.engineering/admin](https://pr-20918.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @rav-redchamps , thanks for the pull request!

Public methods should not be added to classes marked as API without approval by the architect because of backward compatibility concerns.

Please consider introducing a new block class implementing `TabInterface` to avoid adding new public methods to an API class?

**Comment by rav-redchamps** [comment]
@sivaschenko make sense. Will do it by tomorrow via introducing a new block class implementing TabInterface

**Comment by rav-redchamps** [comment]
@sivaschenko just pushed the changes. Please have a look

File: app/code/Magento/Sales/view/base/ui_component/customer_form.xml
Comment (ihor-sviziev) [inline app/code/Magento/Sales/view/base/ui_component/customer_form.xml]: I guess it should be unders ACL, similar to "view orders". Could you update it?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/view/base/ui_component/customer_form.xml
Comment (rav-redchamps) [reply app/code/Magento/Sales/view/base/ui_component/customer_form.xml]: Done. Please check.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-4126](https://jira.corp.magento.com/browse/ENGCOM-4126) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
✔️ QA passed

**Comment by ihor-sviziev** [comment]
Hi @sivaschenko,
Could you clarify why this PR was moved to "On hold"?

File: app/code/Magento/Sales/Block/Adminhtml/ShoppingCartsTab.php
Comment (sivaschenko) [inline app/code/Magento/Sales/Block/Adminhtml/ShoppingCartsTab.php]: The registry is a deprecated class, please use another way to retrieve the current customer id. I.e. passing through an introduced service class.

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @rav-redchamps , thanks for your patience! Unfortunately, we couldn't approve the newly introduced usage of deprecated Registry. Can you please replace it with i.e. service class storing the current customer id.

**Comment by sivaschenko** [comment]
@ihor-sviziev the pull request was put on hold as I needed to discuss and decide on whether the registry usage, in this case, can be accepted

File: app/code/Magento/Sales/Block/Adminhtml/ShoppingCartsTab.php
Comment (sivaschenko) [reply app/code/Magento/Sales/Block/Adminhtml/ShoppingCartsTab.php]: For example, introduce a class that will init customer id from the request and provide a method to retrieve this customer id.

**Comment by p-bystritsky** [comment]
@sivaschenko PR updated, please review.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-4126](https://jira.corp.magento.com/browse/ENGCOM-4126) has been created to process this Pull Request

**Review by sivaschenko — APPROVED** [review]

**Comment by makkoff** [comment]
:heavy_check_mark: QA Passed

**Comment by nelero** [comment]
hi,
any news on it ? can we use the PR or still buggy ?
Thanks

**Pull request merged**
